sdk/state/state_test.go

```diff
@@ -391,6 +414,177 @@ func Test(t *testing.T) {
+
+		var fullySigned bool
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.False(t, fullySigned)
+
+		// Sender: re-confirms P_i by signing D_i and sending back
+		payment, fullySigned, err = sendingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+		ci, di, err := sendingChannel.PaymentTxs(payment)
+		require.NoError(t, err)
+		ci, err = ci.AddSignatureDecorated(payment.CloseSignatures...)
+		require.NoError(t, err)
+		di, err = di.AddSignatureDecorated(payment.DeclarationSignatures...)
+		require.NoError(t, err)
+	}
+
+	// Coordinated Close
+	t.Log("Begin coordinated close process ...")
+	t.Log("Initiator submitting latest declaration transaction")
+	lastD, _, err := initiatorChannel.CloseTxs()
+	require.NoError(t, err)
+	lastD, err = lastD.AddSignatureDecorated(initiatorChannel.LatestCloseAgreement().DeclarationSignatures...)
+	require.NoError(t, err)
+
+	fbtx, err := txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+		Inner:      lastD,
+		FeeAccount: initiator.KP.Address(),
+		BaseFee:    txnbuild.MinBaseFee,
+	})
+	require.NoError(t, err)
+	fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+	require.NoError(t, err)
+	_, err = client.SubmitFeeBumpTransaction(fbtx)
+	require.NoError(t, err)
+
+	t.Log("Initiator proposes a coordinated close")
+	cc, err := initiatorChannel.ProposeCoordinatedClose(0, 0)
+	require.NoError(t, err)
+	cc, err = responderChannel.ConfirmCoordinatedClose(cc)
+	require.NoError(t, err)
+	cc, err = initiatorChannel.ConfirmCoordinatedClose(cc)
+	require.NoError(t, err)
+
+	t.Log("Initiator closing channel with new coordinated close transaction")
+	txCoordinated, err := initiatorChannel.CoordinatedCloseTx()
+	require.NoError(t, err)
+	txCoordinated, err = txCoordinated.AddSignatureDecorated(initiatorChannel.CoordinatedClose().CloseSignatures()...)
+	fbtx, err = txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+		Inner:      txCoordinated,
+		FeeAccount: initiator.KP.Address(),
+		BaseFee:    txnbuild.MinBaseFee,
+	})
+	require.NoError(t, err)
+	fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+	require.NoError(t, err)
+	_, err = client.SubmitFeeBumpTransaction(fbtx)
+	if err != nil {
+		hErr := horizonclient.GetError(err)
+		t.Log("Submitting Close:", txCoordinated.SourceAccount().Sequence, "Error:", err)
+		t.Log(hErr.ResultString())
+	}
+	t.Log("Coordinated close successful")
+
+	// check final escrow fund amounts are correct
+	accountRequest := horizonclient.AccountRequest{AccountID: responder.Escrow.Address()}
+	responderEscrowResponse, err := client.AccountDetail(accountRequest)
```

Comment:
just noticed that we are making a call to horizon. It is best if tests can avoid making calls to an external network, because we don't know if that network will be available during the running of this test.

do we need to make this call or can we check if the value is equal to what we should have initialized it to in the beginning of the test, without making the call to horizon?

Reply:
for this test file we're doing an integration test with a standalone network that includes CAP-21. This makes sure the transactions are formatted correct with correct ops/values and can be submitted successfully, among other things like making sure sequence numbers are correct.

We have a [ticket](https://github.com/stellar/experimental-payment-channels/issues/70) for adding unit tests, which won't depend on a network call